### DISCUSSION TITLE
How to use time builtin in Cirrus Starlark

### DISCUSSION THREAD

**Discussion by nbyavuz** [cirruslabs/cirrus-ci-docs#1173] [Q&A] [answered]
Hi, 

I found that we can use couple of Starlark [builtins](https://cirrus-ci.org/guide/programming-tasks/#builtins) in Cirrus, for example:

``` starlark
load("cirrus", "fs")
```

Is there a way to use [time](https://github.com/qri-io/starlib/tree/master/time) builtin in Cirrus or should I create a feature request for that?

Thanks!

**Comment by fkorotkov** [comment]
What is your use case for time? 

Philosophy behind Starlark is to eliminate side effect and make the execution of the scripts reproducible based on inputs. That's why there is no random and time. Even an iteration over a set is deterministic. Right now the only side effect you can bring to the scripts is via `http` module which we added for calling external APIs mostly.

**Reply by nbyavuz** [reply]
I want to add current time to environment variables, so I can use same time variable from all tasks. Something like:
```starlark
load("cirrus", "env", "time")

def main():
    additional_env = {}

    additional_env['TIME_NOW'] = time.now().unix

    return [
      ('env', additional_env),
    ]
```

**Reply by fkorotkov** [reply, marked as answer]
What if we add `CIRRUS_CHANGE_TIMESTAMP` environment variable instead to expose timestamp of the Git commit?
